runtime/calamari/src/fee.rs

```diff
@@ -125,133 +65,119 @@ mod multiplier_tests {
-        let remark = Call::System(frame_system::Call::<Runtime>::remark_with_event {
-            remark: vec![1, 2, 3],
-        });
-        let len: u32 = remark.encode().len() as u32;
-        let remark_weight: Weight =
-            <Runtime as frame_system::Config>::SystemWeightInfo::remark(len);
-        let max_number_of_remarks_per_block = (block_weight / remark_weight) as u128;
-        let len_fee = max_number_of_remarks_per_block.saturating_mul(
-            <Runtime as pallet_transaction_payment::Config>::LengthToFee::weight_to_fee(
-                &(len as Weight),
-            ),
-        );
+        let mut blocks = 0;
+        let mut fees_paid = 0;
+        let mut fees_after_one_day = 0;
 
-        let base_fee = max_number_of_remarks_per_block
-            * <Runtime as pallet_transaction_payment::Config>::WeightToFee::weight_to_fee(
-                &runtime_common::ExtrinsicBaseWeight::get(),
-            );
+        let info = DispatchInfo {
+            weight: block_weight,
+            ..Default::default()
+        };
 
-        run_with_system_weight(block_weight, || {
-            // initial value configured on module
-            let mut fee_adjustment = Multiplier::one();
-            assert_eq!(fee_adjustment, TransactionPayment::next_fee_multiplier());
-            let mut accumulated_fee: u128 = 0;
-            // Simulates 1 day of parachain blocks (12 seconds each)
-            for iteration in 0..7200 {
-                let next = runtime_multiplier_update(fee_adjustment);
-                // if no change or less, panic. This should never happen in this case.
-                if fee_adjustment >= next {
-                    println!("final fee_adjustment: {fee_adjustment}");
-                    println!("final next: {next}");
-                    panic!("The fee should ever increase");
-                }
-                fee_adjustment = next;
-                let fee =
-                    <Runtime as pallet_transaction_payment::Config>::WeightToFee::weight_to_fee(
-                        &block_weight,
-                    );
+        let mut t: sp_io::TestExternalities = frame_system::GenesisConfig::default()
+            .build_storage::<Runtime>()
+            .unwrap()
+            .into();
+        // set the minimum
+        t.execute_with(|| {
+            pallet_transaction_payment::NextFeeMultiplier::<Runtime>::set(MinimumMultiplier::get());
+        });
+
+        let mut should_fail = false;
+        while multiplier <= Multiplier::from_u32(1) {
+            t.execute_with(|| {
+                // Give the attacker super powers to not pay tx-length fee
+                // The maximum length fo a block is 3_670_016 on Calamari
+                let len = 0;
+                // Imagine this tx was called. This means that he will pay a single base-fee,
+                // which is additional super powers for the attacker
+                let fee = TransactionPayment::compute_fee(len, &info, 0);
```

Comment:
We're giving the attacker some super powers to pay almost no length and base fees.